### PR TITLE
removed unneeded dependencies from pom.xml

### PR DESCRIPTION
**Pull request by phschon**
As discussed in #4

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -85,22 +53,28 @@
             <groupId>org.opentest4j</groupId>
             <artifactId>opentest4j</artifactId>
             <version>1.0.0</version>
+            <scope>test</scope>
```
Comment (matyesz) [inline pom.xml]: Opentest is a transitive dspendency through othed junit dependencies.

**Comment by phschon** [comment]
Opentest was not needed, but I can run the tests without adding further dependencies.
Updated the branch.

**Comment by sbondorf** [comment]
After reducing the explicit JUnit5 dependencies, I cannot start the tests in Eclipse anymore. JUnit5 integration in Ecplise seems very rough at the moment, probably cannot handle transitive dependencies at all:

java.lang.NoClassDefFoundError: org/junit/platform/launcher/core/LauncherFactory
   at org.eclipse.jdt.internal.junit5.runner.JUnit5TestLoader.<init>(JUnit5TestLoader.java:31)
   ...
java.lang.NoClassDefFoundError: org/junit/platform/launcher/core/LauncherFactory
   ...

**Comment by matyesz** [comment]
For eclipse compatibility you have to add (unfortunatelly it does not support junit5 fully at the moment so you need an old launcher that is not transitive from junit5):
<dependency>
            <groupId>org.junit.platform</groupId>
            <artifactId>junit-platform-launcher</artifactId>
            <version>1.1.0</version>
            <scope>test</scope>
        </dependency>

        <dependency>
            <groupId>org.junit.platform</groupId>
            <artifactId>junit-platform-suite-api</artifactId>
            <version>1.1.0</version>
            <scope>test</scope>
        </dependency>

Also we could run mvn clean versions:use-latest-releases to have the latest versions of all our dependencies. As far as I know the open dependencies like [1.1.0,) do not work well.

**Comment by sbondorf** [comment]
Adding the junit-platform-launcher dependency did the trick for Eclipse.

Regarding always using the newest version of dependencies:
In general, the DiscoDNC "core" does not depend on external libraries at all, only the rational number backends (hopefully moved to an extension in the future), the MPA RTC curve backend and the tests do. Therefore, I think that we actually do not gain much from dependency updates. Rather, I feel would loose more in terms of incompatibilities when a newer major version of a dependency breaks the API (remembers me of our migration from JUnit4 to 5). Thus, I am not in favor of given a wild card for all future versions of a dependency.

Instead, I checked the documentation detailing version intervals [1] (thanks for hinting at this feature).  I replaced
* org.junit.jupiter's 5.1.0 with (5.0.0-RC3,5.1],
* org.junit.platform's 1.1.0 with (1.0.0-RC3,1.1], and 
* org.apache.commons, commons-math3's 3.6.1 with [3.0,3.6.1].

I.e., working with ranges closed to the right works for me.

[1] https://docs.oracle.com/middleware/1212/core/MAVEN/maven_version.htm#CJHDEHAB

**Comment by matyesz** [comment]
Do we really want to have Release Candidates in the dependency list? We usually try to avoid that...

**Comment by sbondorf** [comment]
I think you are right, I was too lazy to exclude intermediate milestones and release candidates that the JUnit team published in the Maven Central Repository. I used closed intervals in case the non-numeric suffixes (-M#, -RC#) cause problems.
Apache Commons Math does not publish such versions and thus remains unchanged.

**Comment by matyesz** [comment]
Merged to master. Thanks.

**Pull request merged**